force-app/main/default/classes/ServiceDeliveryController_TEST.cls

```diff
@@ -10,6 +10,178 @@
+            expectedServiceDeliveries
+        );
+        selectorStub.assertCalledWith(
+            'getServiceDeliveriesForSession',
+            Id.class,
+            sessionId
+        );
+    }
+
+    @IsTest
+    private static void testGetNumberOfServiceDeliveriesForSessionException() {
+        Id sessionId = TestUtil.mockId(ServiceSession__c.SObjectType);
+        selectorStub.withThrowException(
+            'getNumberOfServiceDeliveriesForSession',
+            Id.class
+        );
+
+        Test.startTest();
+        ServiceDeliveryController.deliverySelector = (ServiceDeliverySelector) selectorStub.createMock();
+        Exception actualException;
+        try {
+            final Integer actualNumberOfDeliveries = ServiceDeliveryController.getNumberOfServiceDeliveriesForSession(
+                sessionId
+            );
+        } catch (Exception e) {
+            actualException = e;
+        }
+        Test.stopTest();
+
+        System.assertEquals(
+            selectorStub.testExceptionMessage,
+            actualException.getMessage(),
+            'Expected the controller to rethrow the exception from the selector.'
+        );
+
+        selectorStub.assertCalledWith(
+            'getNumberOfServiceDeliveriesForSession',
+            Id.class,
+            sessionId
+        );
+    }
+
+    @IsTest
+    private static void testDeleteServiceDeliveriesForSession() {
+        List<ServiceDelivery__c> expectedServiceDeliveries = new List<ServiceDelivery__c>{
+            new ServiceDelivery__c(
+                Name = 'Test1',
+                Id = TestUtil.mockId(ServiceDelivery__c.SObjectType)
+            ),
+            new ServiceDelivery__c(
+                Name = 'Test2',
+                Id = TestUtil.mockId(ServiceDelivery__c.SObjectType)
+            ),
+            new ServiceDelivery__c(
+                Name = 'Test3',
+                Id = TestUtil.mockId(ServiceDelivery__c.SObjectType)
+            )
+        };
+        Id sessionId = TestUtil.mockId(ServiceSession__c.SObjectType);
+        selectorStub.withReturnValue(
+            'getServiceDeliveriesForSession',
+            Id.class,
+            expectedServiceDeliveries
+        );
+
+        domainStub.withReturnValue(
+            'deleteServiceDeliveries',
+            List<ServiceDelivery__c>.class,
+            null
+        );
+
+        Test.startTest();
+        ServiceDeliveryController.deliverySelector = (ServiceDeliverySelector) selectorStub.createMock();
+        ServiceDeliveryController.deliveryDomain = (ServiceDeliveryDomain) domainStub.createMock();
+        final Integer actualNumberDeleted = ServiceDeliveryController.deleteServiceDeliveriesForSession(
+            sessionId
+        );
+        Test.stopTest();
+
+        System.assertEquals(
+            expectedServiceDeliveries.size(),
+            actualNumberDeleted,
+            'The controller should return that the expected count of deliveries were deleted'
```

Comment:
Should this be reworded: 'The controller should return the expected count of deliveries deleted'. If you think otherwise I am ok with it. This is not a blocker